lms/djangoapps/course_wiki/middleware.py

```diff
@@ -79,7 +79,7 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
                     return redirect('about_course', course_id.to_deprecated_string())
 
                 # If we need enterprise data sharing consent for this course, then redirect to the form.
-                consent_url = get_enterprise_consent_url(request, course_id)
+                consent_url = get_enterprise_consent_url(request, unicode(course_id))
```

Comment:
I see that the use of `to_deprecated_string` above is, well, [deprecated](https://github.com/edx/opaque-keys/blob/ffb2ee1d1be8bb051c32a538067743f74d8503f4/opaque_keys/edx/locator.py#L352). Any reason we shouldn't change the above usages of it now since we're in the function?

Reply:
I'm not entirely sure what's going with `course_path` above. There may be a reason to use the slash-separated style of the course ID. I'd prefer to not mess with that code right now. We could investigate further when we follow up with tests for this un-tested code.

Reply:
:+1: